framework/src/utils/MooseMeshUtils.C

```diff
@@ -144,4 +144,24 @@ getSubdomainIDs(const libMesh::MeshBase & mesh, const std::vector<SubdomainName>
 
   return ids;
 }
+
+bool
+isBoundaryID(const BoundaryName & boundary_name)
+{
+  std::istringstream ss(boundary_name);
+  BoundaryID id;
+  return !(!(ss >> id));
+}
+
+BoundaryID
+getBoundaryID(const BoundaryName & boundary_name, const MeshBase & mesh)
+{
+  BoundaryID id = Moose::INVALID_BOUNDARY_ID;
+  std::istringstream ss(boundary_name);
+
+  if (!(ss >> id))
```

Comment:
Why do this, just check the name?

Reply:
This is used to check whether or not the ID is passed in as an ID or as a name. i.e., a number or a string. In the generator, we need to check explicitly whether it's an ID or not, we can't just skip and look for the ID associated with the name.

Reply:
I just moved this into the rename class... it's use is more clear there.